source/draw/RasterCanvas.ooc

```diff
@@ -88,4 +88,42 @@ RasterCanvas: abstract class extends Canvas {
 			}
 		}
 	}
+	resizeBilinear: static func <T> (sourceBuffer, resultBuffer: T*, sourceBox, resultBox: IntBox2D, sourceStride, resultStride, bytesPerPixel: Int) {
+		(resultWidth, resultHeight) := (resultBox size x, resultBox size y)
+		(sourceWidth, sourceHeight) := (sourceBox size x, sourceBox size y)
+		sourceStartColumn := sourceBox leftTop x
+		sourceStartRow := sourceBox leftTop y
+		resultStartColumn := resultBox leftTop x
+		resultStartRow := resultBox leftTop y
+		for (row in 0 .. resultHeight) {
+			sourceRow := ((sourceHeight as Float) * row) / resultHeight + sourceStartRow
+			sourceRowUp := sourceRow floor() as Int
+			weightDown := sourceRow - sourceRowUp as Float
+			sourceRowDown := (sourceRow - weightDown) as Int + 1
+			if (sourceRowDown >= sourceHeight)
+				weightDown = 0.0f
+			for (column in 0 .. resultWidth) {
+				sourceColumn := ((sourceWidth as Float) * column) / resultWidth + sourceStartColumn
+				sourceColumnLeft := sourceColumn floor() as Int
+				weightRight := sourceColumn - sourceColumnLeft as Float
+				if (sourceColumnLeft + 1 >= sourceWidth)
+					weightRight = 0.0f
+				This _blendSquare(sourceBuffer as UInt8*, resultBuffer as UInt8*, sourceStride, resultStride, sourceRowUp, sourceColumnLeft, row + resultStartRow, column + resultStartColumn, (1.0f - weightDown) * (1.0f - weightRight), (1.0f - weightDown) * weightRight, weightDown * (1.0f - weightRight), weightDown * weightRight, bytesPerPixel)
+			}
+		}
+	}
+	_blendSquare: static func (sourceBuffer, resultBuffer: UInt8*, sourceStride, resultStride, sourceRow, sourceColumn, row, column: Int, weightTopLeft, weightTopRight, weightBottomLeft, weightBottomRight: Float, bytesPerPixel: Int) {
```

Comment:
This method is needed because in ooc you can't call methods on generic types, so you can't use already existing `blend` methods from `Color` classes.
It takes source and result buffer and blends four pixels from source into one pixel in result according to given weights.